### ISSUE TITLE
Mismatch between MiNTLib/FreeMiNT constants

### ISSUE THREAD

**Issue by xdelatour** [freemint/mintlib#57]
I am writing a brief description of SysV functions (pending PR on tos.hyp)

I found a mismatch between the constants used in the MiNTLib and those used in the kernel for the semctl()/Psemctl() `cmd` parameter. 
- MiNTLib uses the same values as glibc and the Linux kernel (11-17)
- FreeMiNT uses the BSD kernel constants (3-9)

The glibc is designed for Linux:
`It officially supports the Hurd and Linux kernels`
A patch is needed for BSD:
`To be compatible with BSD kernel, a lot of patches are applied to the glibc (changes glibc/linux 11-17 to bsd 3-9)`
(source: https://en.wikipedia.org/wiki/Glibc#Supported_hardware_and_kernels)

Therefore, when a program includes sys/sem.h, it sends values between 11 and 17 (glibc/MiNTLib) to the kernel (semctl() just calls Psemctl), but the kernel expects values between 3 and 9 (FreeMiNT/BSD).

Two solutions:
- BSD-patch for the glibc: [sem.h](https://cvs.savannah.nongnu.org/viewvc/glibc-bsd/glibc-bsd/kfreebsd/2.3/sysdeps/unix/bsd/bsd4.4/kfreebsd/bits/sem.h?revision=1.1.1.1&view=markup): values 11-17 (from the glibc) are changed to 3-9 (BSD kernel)
- add glibc/Linux constants 11-17 to the BSD constants 3-9:  [linux_ipc.h](https://github.com/freebsd/freebsd-src/blob/cbf845052f1c2862a64e66dbaf38e887a7592474/sys/compat/linux/linux_ipc.h#L66-L73)

Of course there is no urgency, but hopefully there are no other such problems in the code.

**Comment by mikrosk** [comment]
I presume that none of those are actually implemented? I think it makes sense to change the constants to reflect to FreeMiNT expects.

**Comment by xdelatour** [comment]
> I presume that none of those are actually implemented?
That's right, none are implemented.

It is possible to either modify the MiNTLib to use the kernel constants (3-9), or do the opposite by modifying the kernel to use the MiNTLib constants (11-17)

I don't know what makes the most sense. Anyway these constants are totally unused but they must not be in conflict.

**Comment by mikrosk** [comment]
I'd definitely vote for modifying mintlib. mintlib is glued from various sources, there's no specific reason to maintain glibc compatibility.

**Comment by th-otto** [comment]
I disagree. mintlib is almost execlusively ported from glibc (just a much older version).  Also it could theoretically be the case that some binary was already compiled using the current values, and when you implement them in the kernel using different values, that would give disaster. So if someonwei going to implement Psemctl, those values should be changed first to match mintlib.

Another reason is that, whenever something new is implemented, glibc is usually used as a reference.

**Comment by xdelatour** [comment]
In theory the values could be different but it does not work:
a user program that includes `sys/sem.h` sends mintlib values (GETZCNT=15) while for kernel GETZCNT=7

![bug_consts-glibc_bsd](https://github.com/freemint/mintlib/assets/4384945/c52717b5-448a-4bb7-9128-54676934de32)

I don't have any solution, but we can't maintain mintlib (glibc based), kernel (bsd based) and a patch to make them work together. Even kfreebsd is officially discontinued

I don't plan to implement these functions, but having full documentation of all existing functions will help ensure they work as expected

**Comment by th-otto** [comment]
That's why i suggested to use the mintlibs values also in the kernel. If you use different values, the mintlib's semctl function would have to map them.

But that is rather hypothetical, as long as the kernels Psemctl function does nothing.
